### PR TITLE
Slurm: abstract config preparation

### PR DESCRIPTION
**Pull request by schlad**
A simple move of a config preparation to a separate file.

This is an introduction to extending HPC cluster tests, so certain utility functions will be provided and eventually they might be added to /lib/hpc along with existing hpcbase.pm

Related ticket: https://progress.opensuse.org/issues/51641
Verification run: http://10.160.66.198/tests/2465

### CODE REVIEW

**Comment by okurz** [comment]
If you don't mind please in the description of the PR get rid of the leading spaces in front of "Related ticket" and the following line so that we can click the links :)

**Comment by okurz** [comment]
Instead of a plain IP adress you can also use your DNS-name: f198.suse.de

File: tests/hpc/slurm_config.pm [hunk 1]
```diff
@@ -0,0 +1,36 @@
+# SUSE's openQA tests
+#
+# Copyright © 2017-2018 SUSE LLC
```
Comment (okurz) [inline tests/hpc/slurm_config.pm]: please only use the lines during which you worked on the file, i.e. 2019 I assume :)

Comment (schlad) [reply tests/hpc/slurm_config.pm]: copy&paste syndrome of course. I'll correct it then

**Comment by asmorodskyi** [comment]
1. Mr. Travis is unhappy , please run tidy against your branch 
2. want to see your verification run before merging but machine is unavailable . I suspecting that you will get some perl errors because you using function from other module without declaring export , not sure if it will work ...
3. have you seen https://github.com/os-autoinst/os-autoinst-distri-opensuse/blob/master/lib/hpcbase.pm ? might be another place for function which you created

**Comment by schlad** [comment]
"have you seen https://github.com/os-autoinst/os-autoinst-distri-opensuse/blob/master/lib/hpcbase.pm ? might be another place for function which you created"; sure, I have seen that. After all there is explicit use of that. (use base "hpcbase").
I might eventually move it there, or create another module. For now it felt as not the right place, as the functions there were meant for very general hpc functions. 

As for the remaining 2 points. Sure. I'm on those.

**Comment by schlad** [comment]
@asmorodskyi Please see: 10.160.66.198/tests/3631

**Comment by okurz** [comment]
yes, tests are there to care about ;) As discussed over IRC you need to make sure that the file you want to include is part of the include path, e.g. move slurm_config.pm to "lib/"

**Comment by schlad** [comment]
Changes done as talked with @okurz over IRC. slurm config moved to the /lib/hpcbase.pm

test run:
10.160.66.198/tests/3641

**Comment by okurz** [comment]
Please [squash commits](https://robots.thoughtbot.com/git-interactive-rebase-squash-amend-rewriting-history), rest looks good

**Comment by schlad** [comment]
test run: 10.160.66.198/tests/3651

**Pull request merged**